File: backend/app/schemas/sentence.py
from pydantic import BaseModel
from typing import List, Dict

class SentenceResponse(BaseModel):
    scrambled_words: List[str]
    original_sentence: str
    sentence_id: int
    english_sentence: str
    hints: List[Dict[str, str | int]]

class SubmitSentenceRequest(BaseModel):
    sentence_id: int
    user_answer: str

class SubmitSentenceResponse(BaseModel):
    is_correct: bool
    feedback: str
    translated_sentence: str
    result_id: int
    explanation: str
    sentence_id: int
    user_answer: str

File: frontend/src/types/sentence.ts
export interface SentenceResponse {
  scrambled_words: string[];
  original_sentence: string;
  sentence_id: number;
  english_sentence: string;
  hints: { text: string; usefulness: number }[];
}

export interface SubmitSentenceRequest {
  sentence_id: number;
  user_answer: string;
}

export interface SubmitSentenceResponse {
  is_correct: boolean;
  feedback: string;
  translated_sentence: string;
  result_id: number;
  explanation: string;
  sentence_id: number;
  user_answer: string;
}